### ISSUE TITLE
Send an email when creating an account

### ISSUE THREAD

**Issue by BananeGuimauve** [jhipster/jhipster-dotnetcore#16]
In the Java version, when creating an account, a user receives directly in his mailbox a confirmation email. We want to replicate this behavior in the .Net version. Another question then arises, where to store the templates of the mails ?

**Comment by ivanmonteiro** [comment]
The templates of the emails could be stored as razor templates, the same template files of asp.net core/mvc. 
Maybe the folder could be  Views/Mails or Views/Shared/Mails. 
With this, we can pass a Model object, like WelcomeMailModel, pass it to the view and get the result as the final html for the email.

For reference:
https://www.c-sharpcorner.com/article/send-email-from-a-razor-page-in-net-core-2-using-system-net-mail/
https://corstianboerman.com/2020-01-07/sending-transactional-emails-from-asp-net-core.html

**Comment by nicolas63** [comment]
I think is a good idea :smile: 
Can you try an implementation ?

**Comment by duymap** [comment]
Sending emails is a very common feature and is always needs for any project. So if having it in place, would be a big help for developers. E.g we can have mail sending code function implementation in MailService.cs, it reads config from JSON setting file, and developers replace their STMP information. This has been mentioned in this issue https://github.com/jhipster/jhipster-dotnetcore/issues/1036

**Comment by ivanmonteiro** [comment]
@duymap Yes, It is a very common feature.

I has been a long time since I did the coment above. There are libraries that do exactly what I described such as [FluentEmail](https://github.com/lukencode/FluentEmail). But even a very simple email sending solution like SmtpClient should be enough.

By the way, seems that Microsoft docs do not recommend using SmtpClient in new development due to the lack of modern protocols ([see the full explanation here](https://docs.microsoft.com/en-us/dotnet/api/system.net.mail.smtpclient?view=net-6.0#remarks)) and recommends using [MailKit](https://github.com/jstedfast/MailKit) instead, or other email libraries.

If you're willing to implement this and needs any help or assistance don't be afraid to ask

**Comment by duychungxenera** [comment]
Yes, I'm willing to do it. So to contribute it, want to know process how to contribute this repo ? Create branch and do PR against here or need to create new fork and PR ?

**Comment by ivanmonteiro** [comment]
> Yes, willing to do. Just quick questions to ask for speeding up:
> 
> * What are the templates code located?

Here is the MailService class:
https://github.com/jhipster/jhipster-dotnetcore/blob/main/generators/server/templates/dotnetcore/src/Project.Domain.Services/MailService.cs.ejs

And smtp credentials/configuration shoud be added here:
https://github.com/jhipster/jhipster-dotnetcore/blob/main/generators/server/templates/dotnetcore/src/Project.Infrastructure/Configuration/SecuritySettings.cs.ejs

This loads security settings stored add appsettings.{environment}.json

> * Develop is the latest branch that I should branch off, then create PR ?

It seems that develop is outdated. 
You shoud create a branch from **main** and then create a PR here
